noir/noir-repo/acvm-repo/brillig/src/opcodes.rs

```diff
@@ -65,6 +65,85 @@ pub struct HeapVector {
     pub size: MemoryAddress,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, Copy, PartialOrd, Ord)]
+pub enum IntegerBitSize {
+    U0, // Uninitialized
```

Comment:
```suggestion
```
Why would this ever be used? If we want to state that we have an integer but do not know the bit size yet we should use an `Option<IntegerBitSize>`

Reply:
After reviewing again, is this for a default memory value? We could probably just use a `U1` bit size as the default is just going to be `0` anyway.

Reply:
yes, it's the default memory value :D previously it was BigUInt::zero() and bit_size: 0. We don't want to make it u1 because we want to fail for mismatching bit sizes if a wrongly constructed program tries to add a U1 and a null (u0)

Reply:
Makes sense about failing on mismatching bit sizes. I contend that having an `Option< IntegerBitSize >` is still the better pattern, but as this was the functionality that existed previously I will not block on this.